### ISSUE TITLE
Android build issue: p

### ISSUE THREAD

**Issue by fninsiima** [tensorflow/tensorflow#4399]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
None relating to permission denied issue

### Environment info
Operating System: Linux Ubuntu 64-bit 

If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`)
2. The output of `bazel version`

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
android_sdk_repository(
    name = "androidsdk",
    api_level = 23,
    build_tools_version = "24.0.2",
    # Replace with path to Android SDK on your system
    path = "/home/flavia/Documents/android-sdk-linux",
)
#
android_ndk_repository(
    name="androidndk",
    path="/home/flavia/Documents/ndk",
    api_level=21)

### What other attempted solutions have you tried?
Downloaded NDK version: android-ndk-r12b, unzipped it and renamed it to ndk and pointed to it in workspace
Downloade SDK: 24.4.1, run SDK manager to install build-tools, pointed to the main sdk directory

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).
ERROR: /home/flavia/.cache/bazel/_bazel_flavia/1fe576225e857ae381761d2049465367/external/protobuf/BUILD:71:1: C++ compilation of rule '@protobuf//:protobuf_lite' failed: arm-linux-androideabi-gcc failed: error executing command 
  (cd /home/flavia/.cache/bazel/_bazel_flavia/1fe576225e857ae381761d2049465367/execroot/tensorflow && \
  exec env - \
    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games \
  external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-gcc -fstack-protector-strong -fpic -ffunction-sections -funwind-tables -no-canonical-prefixes -fno-canonical-system-headers '-march=armv7-a' '-mfpu=vfpv3-d16' '-mfloat-abi=softfp' -MD -MF bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/bin/external/protobuf/_objs/protobuf_lite/external/protobuf/src/google/protobuf/repeated_field.d '-frandom-seed=bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/bin/external/protobuf/_objs/protobuf_lite/external/protobuf/src/google/protobuf/repeated_field.o' -iquote external/protobuf -iquote bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/genfiles/external/protobuf -iquote external/bazel_tools -iquote bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/genfiles/external/bazel_tools -isystem external/protobuf/src -isystem bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/genfiles/external/protobuf/src -isystem external/bazel_tools/tools/cpp/gcc3 -DHAVE_PTHREAD -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare '-Wno-error=unused-function' '--sysroot=external/androidndk/ndk/platforms/android-21/arch-arm' -isystem external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9/include -isystem external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9/include-fixed -isystem external/androidndk/ndk/sources/cxx-stl/gnu-libstdc++/4.9/include -isystem external/androidndk/ndk/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi-v7a/include -isystem external/androidndk/ndk/sources/cxx-stl/gnu-libstdc++/4.9/include/backward -c external/protobuf/src/google/protobuf/repeated_field.cc -o bazel-out/android-arm-linux-androideabi-4.9-v7a-gnu-libstdcpp-fastbuild/bin/external/protobuf/_objs/protobuf_lite/external/protobuf/src/google/protobuf/repeated_field.o): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
process-wrapper: execvp("external/androidndk/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-gcc", ...): Permission denied
Target //tensorflow/examples/android:tensorflow_demo failed to build
INFO: Elapsed time: 1.330s, Critical Path: 0.43s

Basically, i've failed to decipher the permission denied. I'm no expert at android, but i'm experimenting with tensorflow.

**Comment by andrewharp** [comment]
hi @fninsiima 

First, can you verify that `/home/flavia/Documents/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-gcc` exists on your system and is executable?

As a potential quick fix, I would suggest that you try downloading/extracting r11c from here: https://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip, as occasionally Bazel will exhibit incompatibilities with the latest NDK.

**Comment by andrewharp** [comment]
Also, did you clone with `--recurse-submodules`, and can you verify that `<git_root>/tensorflow/bazel-tensorflow/external/protobuf` is not empty?

**Comment by fninsiima** [comment]
Bazel protobuf is not empty. Ive listed the output here

flavia@fninsiima-ThinkPad-T440p:~/Documents/tensorflow/bazel-tensorflow/external/protobuf$ ls
appveyor.bat      generate_descriptor_proto.sh  protobuf-lite.pc.in
appveyor.yml      gmock.BUILD                   protobuf.pc.in
autogen.sh        java                          Protobuf.podspec
benchmarks        javanano                      protoc-artifacts
BUILD             jenkins                       python
CHANGES.txt       js                            README.md
cmake             LICENSE                       ruby
configure.ac      m4                            six.BUILD
conformance       Makefile.am                   src
CONTRIBUTORS.txt  more_tests                    tests.sh
csharp            objectivec                    update_file_lists.sh
docs              php                           util
editors           post_process_dist.sh          WORKSPACE
examples          protobuf.bzl

Yes, ive verified that arm-linux-androideabi-gcc  exists on my system.
But I'm not sure how to verify that its executable...
On opening it(double clicking), it says 
---
'There is no application installed for “executable” files.'
---
On doing so in terminal, it says
------
The program 'arm-linux-androideabi-gcc' is currently not installed. You can install it by typing:
sudo apt-get install gcc-arm-linux-androideabi
------

No, i do not recall cloning with --recurse-submodules.

**Comment by fninsiima** [comment]
No, i have yet to try out the quick fix.

**Comment by andrewharp** [comment]
```
Yes, ive verified that arm-linux-androideabi-gcc exists on my system.
But I'm not sure how to verify that its executable...
```

You would just run `ls -la /home/flavia/Documents/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-gcc` and make sure you have read/execute permissions, or simply try executing it directly from the command line.

**Comment by fninsiima** [comment]
Okay.

I followed your instructions.
Then later on i run into an issue. something to do with 

arm-linux-androideabi-gcc: error trying to exec 'cc1': execvp: No such file...

Then i google a bit. and found a solution here...

http://stackoverflow.com/questions/10207280/error-compiling-android-on-ubuntu-11-10

In short, i did not extract my ndk with the right tool. so i did not have executable permissions .

So i re-did the extraction using unzip.

Voila! Problems solved with permissions.

SUCCESSFUL BUILD ACCOMPLISHED!

Hope you don't mind if i close this issue.

Thanks very much Andrew. God bless.
